src/Validation/ValidationRule.php

```diff
@@ -111,41 +58,29 @@ public function isLast(): bool
      * @throws \InvalidArgumentException when the supplied rule is not a valid
      * callable for the configured scope
      */
-    public function process(mixed $value, array $providers, array $context = []): array|string|bool
+    public function process(mixed $value, array $context = []): array|string|bool
     {
-        $context += ['data' => [], 'newRecord' => true, 'providers' => $providers];
+        $context += ['data' => [], 'newRecord' => true];
 
         if ($this->_skip($context)) {
             return true;
         }
 
-        if (is_string($this->_rule)) {
-            $provider = $providers[$this->_provider];
-            /** @phpstan-ignore-next-line */
-            $callable = [$provider, $this->_rule](...);
-        } else {
-            $callable = $this->_rule;
-            if (!$callable instanceof Closure) {
-                $callable = $callable(...);
-            }
-        }
-
         $args = [$value];
-
-        if ($this->_pass) {
-            $args = array_merge([$value], array_values($this->_pass));
+        if ($this->pass) {
+            $args = array_merge([$value], array_values($this->pass));
         }
 
-        $params = (new ReflectionFunction($callable))->getParameters();
+        $params = (new ReflectionFunction($this->callable))->getParameters();
         $lastParm = array_pop($params);
         if ($lastParm && $lastParm->getName() === 'context') {
```

Comment:
```suggestion
        $lastParam = array_pop($params);
        if ($lastParam && $lastParam->getName() === 'context') {
```

Small typo we could fix.

Reply:
Done